src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -5831,11 +5840,45 @@ namespace InternalHeatGains {
                 }
             }
             TDPAirIn = PsyTdpFnWPb(WAirIn, StdBaroPress, RoutineName);
-            RHAirIn = PsyRhFnTdbWPb(TAirIn, WAirIn, StdBaroPress, RoutineName);
+            RHAirIn = 100.0 * PsyRhFnTdbWPb(TAirIn, WAirIn, StdBaroPress, RoutineName); // RHAirIn is %
 
             // Calculate power input and airflow
             TAirInDesign = ZoneITEq(Loop).DesignTAirIn;
 
+            Real64 DesignTotalPower = ZoneITEq(Loop).DesignTotalPower;
+            Real64 AirVolFlowAtDesign = ZoneITEq(Loop).DesignAirVolFlowRate;
+
+            if (DoingSizing && ZoneITEq(Loop).FlowControlWithApproachTemps) {
+                SetPointManager::GetSetPointManagerInputData(ErrorsFound);
+                for (int SetPtMgrNum = 1; SetPtMgrNum <= SetPointManager::NumSZClSetPtMgrs; ++SetPtMgrNum) {
+                    if (SetPointManager::SingZoneClSetPtMgr(SetPtMgrNum).ControlZoneNum == Loop) {
+                        TAirInDesign = SetPointManager::SingZoneClSetPtMgr(SetPtMgrNum).MaxSetTemp; // Set the maximun setpoint
+                    }
+                }
+                if (ZoneITEq(Loop).SupplyApproachTempSch != 0) {
+                    TAirInDesign = TAirInDesign + GetCurrentScheduleValue(ZoneITEq(Loop).SupplyApproachTempSch);
+                } else {
+                    TAirInDesign = TAirInDesign + ZoneITEq(Loop).SupplyApproachTemp;
+                }
+                Real64 OperSchedFrac = GetCurrentScheduleValue(ZoneITEq(Loop).OperSchedPtr);
+                Real64 CPULoadSchedFrac = GetCurrentScheduleValue(ZoneITEq(Loop).CPULoadSchedPtr);
+
+                Real64 CPUPowerAtDesign =
+                        max(ZoneITEq(Loop).DesignCPUPower * OperSchedFrac * CurveManager::CurveValue(ZoneITEq(Loop).CPUPowerFLTCurve, CPULoadSchedFrac, TAirInDesign), 0.0);
+
+                AirVolFlowFracDesignT = max(CurveManager::CurveValue(ZoneITEq(Loop).AirFlowFLTCurve, CPULoadSchedFrac, TAirInDesign), 0.0);
+                AirVolFlowAtDesign = ZoneITEq(Loop).DesignAirVolFlowRate * OperSchedFrac * AirVolFlowFracDesignT;
+
+                Real64 FanPowerAtDesign =
+                        max(ZoneITEq(Loop).DesignFanPower * OperSchedFrac * CurveManager::CurveValue(ZoneITEq(Loop).FanPowerFFCurve, AirVolFlowFracDesignT), 0.0);
+
+
+                if (FanPowerAtDesign + CPUPowerAtDesign > ZoneITEq(Loop).DesignTotalPower) {
+                    DesignTotalPower = FanPowerAtDesign + CPUPowerAtDesign;
+                }
+
+            }
+
```

Comment:
Why repeat all of this every sizing timestep here in `CalcZoneITEq` when it's already been done in `GetInternalHeatGainsInput`? 

Seems you could add variables to the `ZoneITEq` struct and save the result there to use here. This may not even require new variables, seems like the adjusted values could be saved once to `ZoneITEq(Loop).DesignTAirIn,  ZoneITEq(Loop).DesignTotalPower, and  ZoneITEq(Loop).DesignAirVolFlowRate`. What's changing here an what's not?

At a minimum, search the setpoint managers in `GetInternalHeatGainsInput` and save the SPM index for use here. And the block in `GetInternalHeatGainsInput` shouldn't be querying current schedule values - that has no meaning during get input. So maybe some of the code there needs to be removed?